src/Ellipsis.js

```diff
@@ -0,0 +1,84 @@
+import React, { Component, createRef } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { throttle } from 'lodash';
+
+class Ellipsis extends Component {
+    containerRef = createRef();
+
+    state = {
+        isPrepared: false,
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.throttleHandleResize = throttle(this.handleResize, 250);
+    }
+
+    componentDidMount() {
+        window.addEventListener('resize', this.throttleHandleResize, { passive: true });
+
+        this.ellipsize();
+
+        this.setState({ isPrepared: true });
+    }
+
+    componentDidUpdate() {
+        this.ellipsize();
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('resize', this.throttleHandleResize, { passive: true });
+    }
+
+    render() {
+        const { children, tag: Tag, className } = this.props;
+        const { isPrepared } = this.state;
+
+        const containerClassName = classNames('react-ellipsis_container', {
+            reveal: isPrepared,
+        }, className);
+
+        return (
+            <Tag
+                ref={ this.containerRef }
+                className={ containerClassName }
+                { ...this.props }>
+                { children }
+            </Tag>
+        );
+    }
+
+    ellipsize = () => {
+        const element = this.containerRef.current;
+        const words = this.props.children;
+        let wordsArray = [];
+
+        element.innerHTML = words;
+
+        wordsArray = typeof words === 'string' && words.split(' ');
+
+        /* istanbul ignore next */
```

Comment:
Yeah, I think this is not easy to test because we won't have the right heights in jest context :+1: